software/bsg_manycore_lib/bsg_tilegroup.h

```diff
@@ -104,42 +104,62 @@ void extern_store_char(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset,
 }
 
 
-float extern_load_float(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset) {
+float extern_load_float(int STRIPE *arr_ptr, unsigned elem_size, unsigned offset,
```

Comment:
just for my understanding, when is extern_load_float used?

Reply:
When loading a float from another tile -- it's effectively the same as extern_load_int, but clang has much stricter typing requirements than GCC

Reply:
okay. so are you saying that calling "A[i][id]" gets translated into extern_load_<type>() by the compiler (in layman's term)?

Reply:
If A[i][id] is an rvalue, then yes! If it's an lvalue, A[i][id] translates into extern_store_() by the LLVM pass.